### PR TITLE
Embargoed: implement skipping of embargoed updates

### PR DESCRIPTION
**Pull request by asmorodskyi**
Embargoed: implement skipping of embargoed updates

### CODE REVIEW

File: openqabot/types/incident.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def __init__(self, incident):
         self.id = incident["number"]
         self.rrid = f"{self.project}:{self.rr}" if self.rr else None
         self.staging = not incident["inReview"]
+        self.embargoed = incident["crd"] is not None
```
Comment (mimi1vx) [inline openqabot/types/incident.py:22]: this is wrong :D It will have set CRD after embargo is lifted ...

So it should also compare dater time in CRD with current date time

Comment (asmorodskyi) [reply openqabot/types/incident.py:22]: I replied to you in private . resolving this here if I not convinced you please reply me in private too

**Review by mimi1vx — CHANGES_REQUESTED** [review]

File: openqabot/types/baseconf.py [hunk 2]
```diff
@@ -24,3 +24,6 @@ def __call__(
     @abstractstaticmethod
     def normalize_repos(config):
         pass
+
+    def filter_embargoed(self) -> bool:
+        return bool(self.settings.get("PUBLIC_CLOUD_RUN", False))
```
[the remark this answers is not in the document]
Comment (mimi1vx) [reply openqabot/types/baseconf.py:29]: +- here with this `return any( k.startswith("PUBLIC") for k in self.settings.keys())` we got the same effect without a new variable to use in the job call :D

**Comment by asmorodskyi** [comment]
DISCLAIMER : I have 0 confidence in changes what I have done and without real run  ( not dry_run ) I can not prove anything . Tests are passing and I assume everything should correct but please be picky with this one to not solve disaster later

**Comment by asmorodskyi** [comment]
beside variable name raised by @mimi1vx  , ** PR is ready for review **. 
( variable name can be changed within 1 minute after I get suggestion what should be the name ... )

File: openqabot/smeltsync.py [hunk 3]
```diff
@@ -116,10 +116,7 @@ def _create_record(cls, inc):
         incident["approved"] = approved
         incident["rr_number"] = rr_number
         incident["inReviewQAM"] = inReviewQAM
-        # this is temporary solution. At first stage we will just push this field into dashboard
-        # next we will add logic to store it in dashboard and then finally we will replace
-        # here dummy value with real one from smelt
-        incident["embargoed"] = False
+        incident["embargoed"] = inc["crd"][0]["crd"]
```
Comment (mimi1vx) [inline openqabot/smeltsync.py:119]: what type is `inc["crd"][0]["crd"]` and what happens if is "crd" missing ?

File: openqabot/types/baseconf.py [hunk 4]
```diff
@@ -1,17 +1,17 @@
 # Copyright SUSE LLC
 # SPDX-License-Identifier: MIT
-from abc import ABCMeta, abstractmethod, abstractstaticmethod
```
Comment (mimi1vx) [inline openqabot/types/baseconf.py:3]: Why ?

File: openqabot/types/baseconf.py [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
 from typing import Any, Dict, List, Optional
 
 from .incident import Incident
 
 
-class BaseConf(metaclass=ABCMeta):
```
Comment (mimi1vx) [inline openqabot/types/baseconf.py:9]: dtto

File: openqabot/types/baseconf.py [hunk 6]
```diff
@@ -21,6 +21,9 @@ def __call__(
     ) -> List[Dict[str, Any]]:
         pass
 
-    @abstractstaticmethod
+    @staticmethod
```
Comment (mimi1vx) [inline openqabot/types/baseconf.py:24]: why this change ? its abstract method which needs to be implemented ..

File: openqabot/types/baseconf.py [hunk 7]
```diff
@@ -1,17 +1,17 @@
...
 # SPDX-License-Identifier: MIT
-from abc import ABCMeta, abstractmethod, abstractstaticmethod
 from typing import Any, Dict, List, Optional
 
 from .incident import Incident
 
 
-class BaseConf(metaclass=ABCMeta):
+class BaseConf:
+    OBEY_EMBARGOED_CONF = "PUBLIC_CLOUD_RUN"
```
Comment (mimi1vx) [inline openqabot/types/baseconf.py:9]: no , this is pretty awful ...

File: openqabot/types/baseconf.py [hunk 8]
```diff
@@ -21,6 +21,9 @@ def __call__(
     ) -> List[Dict[str, Any]]:
         pass
 
-    @abstractstaticmethod
+    @staticmethod
     def normalize_repos(config):
         pass
+
+    def filter_embargoed(self) -> bool:
+        return bool(self.settings.get(BaseConf.OBEY_EMBARGOED_CONF, False))
```
Comment (mimi1vx) [inline openqabot/types/baseconf.py:29]: same problem as before, don use `settings` for bot configuration --> `settings` is mean to be passed into openQA.

**Review by mimi1vx — CHANGES_REQUESTED** [review]

File: openqabot/smeltsync.py (hunk 3 above)
Comment (asmorodskyi) [reply openqabot/smeltsync.py:119]: 1. crd suppose to be boolean 
2. Regarding what will happen if field will be missing I think same question can be applied to ALL fields in `inc` , right ? Don't see why `crd` field should have special treatment here . All fields which we got here if I understand everything right are defined by `INCIDENT` query in smelt.py . So removal of any field from this query will cause exceptions in this function ( and in many other places ) .

File: openqabot/types/baseconf.py (hunk 4 above)
Comment (asmorodskyi) [reply openqabot/types/baseconf.py:3]: Because I need to test `is_embargoed` method , for this I need to create instance of this class see test code

File: openqabot/types/baseconf.py (hunk 5 above)
Comment (asmorodskyi) [reply openqabot/types/baseconf.py:9]: I am closing this comment , it is just duplication of same question as above , don't see the point to copy/pasting same answer two times

File: openqabot/types/baseconf.py (hunk 7 above)
Comment (asmorodskyi) [reply openqabot/types/baseconf.py:9]: why ? please provide some practical reasoning . such comment is applicable to movie that you have watched or food which you tried . In code review it is not allow me to keep meaningful discussion . you saying "this is awful" and I am saying "no this is cool" and we are blocked ...

From my side I can say that such approach allow to avoid typos in variable name when it is used in several places and ease change of its name

File: openqabot/types/baseconf.py (hunk 8 above)
Comment (asmorodskyi) [reply openqabot/types/baseconf.py:29]: ok will change that

File: openqabot/types/baseconf.py (hunk 4 above)
Comment (mimi1vx) [reply openqabot/types/baseconf.py:3]: test can be on Any other class which inherits  from BaseConf ......

File: openqabot/smeltsync.py (hunk 3 above)
Comment (mimi1vx) [reply openqabot/smeltsync.py:119]: 1 ... `crd` according docs is `DateTime` and api of dashboard excepts boolean ...

Comment (mimi1vx) [reply openqabot/smeltsync.py:119]: 1) `crd`  according to docs is `DateTime`  ...

File: openqabot/types/baseconf.py (hunk 4 above)
Comment (asmorodskyi) [reply openqabot/types/baseconf.py:3]: ok will do that

File: openqabot/types/baseconf.py (hunk 7 above)
Comment (asmorodskyi) [reply openqabot/types/baseconf.py]: resolved

File: openqabot/types/baseconf.py (hunk 4 above)
Comment (asmorodskyi) [reply openqabot/types/baseconf.py]: done

File: openqabot/loader/smelt.py [hunk 9]
```diff
@@ -25,7 +25,8 @@
 INCIDENT = '{incidents(incidentId: %(incident)s) { edges { node {emu project \
 repositories { edges { node { name } } } requestSet(kind: "RR") { edges { node \
 { requestId status { name } reviewSet { edges { node { assignedByGroup { name } \
-status { name } } } } } } } packages { edges { node { name } } } } } } }'
+status { name } } } } } } } packages { edges { node { name } } } } } \
+    edges{ node { crd } } } }'
```
Comment (mimi1vx) [inline openqabot/loader/smelt.py:29]: please update also `INCIDENT_SCHEMA`

Comment (asmorodskyi) [reply openqabot/loader/smelt.py:29]: can you help me with that ?

File: openqabot/smeltsync.py (hunk 3 above)
Comment (asmorodskyi) [reply openqabot/smeltsync.py]: during initial discussions I remember statement that we can not rely on this date , because actual agreement saying " until customer get the update first " which normally happening before date defined in crd but not always and in THAT case we should obey actual delivery not this date . 

So agreement which I remember was that **everything** what has this date should be blocked no matter if date there in the past or future . Just the fact of something defined in this field give us true

File: tests/test_smeltsync.py [hunk 10]
```diff
@@ -90,7 +90,7 @@ def reply_callback(request):
 @responses.activate
 @pytest.mark.parametrize("fake_qem", [()], indirect=True)
 @pytest.mark.parametrize(
-    "fake_smelt_api", [(["qam-openqa", "new", "review"])], indirect=True
+    "fake_smelt_api", [(["qam-openqa", "new", "review", True])], indirect=True
```
[the remark this answers is not in the document]
Comment (asmorodskyi) [reply tests/test_smeltsync.py]: fixed

File: openqabot/loader/smelt.py (hunk 9 above)
Comment (asmorodskyi) [reply openqabot/loader/smelt.py:29]: so I added something , not sure if it , right

**Review by mimi1vx — APPROVED** [review]
LGTM

**Review by okurz — APPROVED** [review]
